src/phasorpy/flim.py

```diff
@@ -0,0 +1,498 @@
+
+The ``phasorpy.flim`` module provides functions and classes for the calibration and analysis
+of time-resolved fluorescence lifetime imaging (FLIM) data using phasor techniques. The module
+is designed to facilitate the processing and interpretation of FLIM data, including the correction
+of phase and modulation parameters and the application of phasor analysis.
+
+Classes
+-------
+FlimData
+    Represents fluorescence lifetime imaging (FLIM) data with calibration capabilities.
+    - Attributes:
+        - `laser_frequency`
+        - `original_data`
+        - `average_phasor`
+        - `real_phasor_coordinates`
+        - `imaginary_phasor_coordinates`
+        - `axis`
+        - `phi_correction`
+        - `modulation_correction`
+        - `calibration_status`: 
+
+    - Methods:
+        - `calibrate_flim`
+        
+Functions
+---------
+calibrate_multiple_flim
+    Calibrates a list of time-resolved FLIM data using reference with known lifetime, or with phase and modulation correction parameters.
+
+Constants
+---------
+TAU_REFERENCES: dict
+    Lifetime data of selected fluorophores to use as references for calibration.
+    Data is obtained from the ISS website: https://iss.com/resources#lifetime-data-of-selected-fluorophores.
+
+"""
+
+from __future__ import annotations
+
+__all__ = [
+    'FlimData',
+    'TAU_REFERENCES',
+]
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ._typing import Any, NDArray, Callable, Dict, Sequence
+
+import numpy
+from numpy.testing import assert_array_equal
+
+from phasorpy.phasor import phasor
+
+
+class FlimData:
+    """
+    Represents fluorescence lifetime imaging (FLIM) data with calibration capabilities.
+
+    Parameters
+    ----------
+    laser_frequency : float
+        The frequency of the laser used for data acquisition.
+    original_data : numpy.ndarray
+        The original FLIM data.
+    average_phasor : numpy.ndarray
+        The average image along the specified axis
+    real_phasor_coordinates : numpy.ndarray
+        The real component of the fft transformed FLIM data
+    imaginary_phasor_coordinates : numpy.ndarray
+        The imaginary component of the fft transformed FLIM data
+    axis : int, optional
+        The axis along which phasor calculations are performed (default is 0).
+    phi_correction : float, optional
+        The initial correction applied to the phase angle (default is 0).
+    modulation_correction : float, optional
+        The initial correction applied to the modulation amplitude (default is 1).
+    calibration_status : bool, optional
+        The initial calibration status (default is False).
+
+    Methods
+    -------
+    calibrate_flim(recalibrate=False, reference_data=None, reference_name=None, reference_tau=None,
+                   center_function=None, **kwargs)
+        Calibrates the FlimData object based on reference data and correction parameters.
+
+    Attributes
```

Comment:
Public class attributes should be typed and can be documented by inline docstrings I think. That avoids duplication.